content/articles/setup-and-run-applications-using-typescript-with-aureliajs/index.md

```diff
@@ -0,0 +1,270 @@
+
+Head over to your project directory and create an `app.ts` file to create an interface. Then use the keyword `interface` followed by your interface name, as shown below.
+
+```ts
+interface Todo {
+}
+```
+
+The add properties that define the shape of the object as shown below;
+
+```ts
+interface Todo {
+  title: string;
+  description: string
+  completed: boolean;
+}
+```
+
+Any object that doesn't have this described structure is rendered incompatible. The consumers of this interface must implement these properties.
+
+This becomes quite useful when you have multiple teams working on a very large project. If someone decides to name things differently or forget something, the application will adhere to the set properties. The properties named differently will automatically become incompatible with the set object. This helps you avoid some code errors around your teams.
+
+Therefore, it makes sense to have a common description of the specific properties of a todo.
+
+### Create todos class
+
+As described above, a class acts as a blueprint for an object. The object is now created in the above step. Let's now set up a class that describes its blueprint. To do this, Navigate to the `app.ts` file and follow the following steps;
+
+### Set up the `App` class
+
+One of the important use cases where an interface is really useful is when setting up a class. This sets the compatibility of your data. A class basically has the same properties as the `interface` at any given time. If all the required properties are not set, the whole data set becomes incompatible. Below is how the class `App` will be set based on the interface `Todo`.
+
+```ts
+class App {
+  heading = "Todos List APP";
+  todos: Todo[] = [];
+  todoTitle = '';
+  todoDescription = '';
+}
+```
+
+This will add a heading/ title to the application and array of todos and empty properties `todoTitle` and `todoDescription`.
+
+### Create constructors and methods
+
+A class goes hand in hand with a constructor. A constructor is commonly referred to as a special method that has the same name as the class. In Typescript, a constructor is a special function with the keyword `constructor`. It constructs/initializes the object before it gets assigned to its methods.
+
+A method gets assigned to the initialized constructor. Here you will create methods to add, remove and clone a todo. The methods body specifies the properties you want to want access. These are the properties of the class on which you're running this method. In simple terms, a method runs an instance of the class `App`.
+
+Below is how to set up these constructors and methods. Thet goes inside the class `App`:
+
+- Add a new todo
+
+```ts
+addTodo() {
+    if (this.todoTitle || this.todoDescription) {
+        this.todos.push({
+            title: this.todoTitle,
+            description: this.todoDescription,
+            completed: false
+        });
+        this.todoTitle = '';
+        this.todoDescription = '';
+    }
+}
+```
```

Comment:
Can you explain the codes line-by-line below as bulleted points?

Do the same for the below code snippets too